iree/hal/vulkan/vulkan_device.cc

```diff
@@ -642,60 +642,89 @@ StatusOr<ref_ptr<Event>> VulkanDevice::CreateEvent() {
   return make_ref<NativeEvent>(add_ref(logical_device_), event_handle);
 }
 
-StatusOr<ref_ptr<BinarySemaphore>> VulkanDevice::CreateBinarySemaphore(
-    bool initial_value) {
-  IREE_TRACE_SCOPE0("VulkanDevice::CreateBinarySemaphore");
+StatusOr<ref_ptr<Semaphore>> VulkanDevice::CreateSemaphore(
+    uint64_t initial_value) {
+  IREE_TRACE_SCOPE0("VulkanDevice::CreateSemaphore");
+
+  VkSemaphoreTypeCreateInfo timeline_create_info;
+  timeline_create_info.sType = VK_STRUCTURE_TYPE_SEMAPHORE_TYPE_CREATE_INFO;
+  timeline_create_info.pNext = nullptr;
+  timeline_create_info.semaphoreType = VK_SEMAPHORE_TYPE_TIMELINE;
+  timeline_create_info.initialValue = initial_value;
```

Comment:
Relating to my prior comment about `NativeTimelineSemaphore` / `EmulatedTimelineSemaphore`, this could be moved e.g. `static StatusOr<NativeTimelineSemaphore> NativeTimelineSemaphore::CreateSemaphore(args...)`

(Separation of concerns + more flexibility in emulating if needed, though that *might* be YAGNI... hoping to test on Android soon to see if the extension layer will be sufficient)

Reply:
Good thinking! Done.